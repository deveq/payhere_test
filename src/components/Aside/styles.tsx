import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;