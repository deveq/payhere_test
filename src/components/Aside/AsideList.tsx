import { Dispatch, SetStateAction } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import * as Styled from './styles';
import {useRecoilState} from 'recoil';
import {asideSelectedIndex} from 'atoms/asideSelectedIndexState';

interface AsideListProps {
	data: string[];
	index: number;
	setIndex: Dispatch<SetStateAction<number>>;
}
export interface IAsideListItem {
	title: string;
}

const AsideList = ({data, index, setIndex}: AsideListProps) => {
	return (
		<>
			<Styled.List>
				{data.map((item, itemIndex) => (
					<Styled.ListItem
                        key={item}
						onClick={() => {
							setIndex(itemIndex);
						}}
						selected={index === itemIndex}
					>
						{item}
					</Styled.ListItem>
				))}
			</Styled.List>
		</>
	);
};

export default AsideList;
