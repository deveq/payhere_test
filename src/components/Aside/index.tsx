import * as Styled from './styles';
import AsideList, {IAsideListItem} from './AsideList';

interface AsideProps {
    selectedIndex: number;
    data: IAsideListItem[];
}

const Aside = ({selectedIndex, data}: AsideProps) => {
    return (
        <Styled.Aside>
            <Styled.Title>설정</Styled.Title>
            <AsideList selectedIndex={selectedIndex} data={data} />
        </Styled.Aside>
    )
}

export default Aside;