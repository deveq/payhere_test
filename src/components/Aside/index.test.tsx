import {useState} from 'react';
import {render} from '@testing-library/react';
import {renderHook} from '@testing-library/react-hooks';
import 'jest-styled-components';
import {renderWithWrapper} from 'tests';
import Aside from './index';

// interface AsideProps {
//     data: string[];
//     title: string;
//     setIndex: Dispatch<SetStateAction<number>>;
//     index: number;
// }

describe('<Aside />', () => {
    const data = ['a', 'b', 'c', 'd'];
    const titleText = 'title'
    it('정상적으로 렌더링되는지 확인한다.', () => {
        const {result: {current: [index, setIndex]}} = renderHook(() => useState(0))
        const {container} = renderWithWrapper(<Aside 
            data={data}
            title={titleText}
            setIndex={setIndex}
            index={index}
        />);

        const aside = container.children[0];
        expect(aside).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    })
})

// export const Aside = styled.aside`
//     background: ${({theme}) => theme.bg_page1};
// `;

// export const List = styled.ul`
// `;

// export const Title = styled.h2`
//     border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
//     color: ${({theme}) => theme.text1};
// `;

// export const ListItem = styled.li<{selected: boolean}>`
//     background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
//     color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
//     &:hover,
//     &:active {
//         background: ${({theme}) => theme.bg_element1};
//     };
// `;

// export const Subtitle = styled.div`

// `;