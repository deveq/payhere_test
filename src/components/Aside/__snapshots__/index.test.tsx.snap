// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aside /> 정상적으로 렌더링되는지 확인한다. 1`] = `
.c0 {
  width: 200px;
  background: #F8F9FA;
  padding: 30px;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 {
  display: grid;
  gap: 5px;
}

.c1 {
  font-size: 24px;
  font-weight: bold;
  border-bottom: 1px solid #DEE2E6;
  color: #212529;
  padding-bottom: 10px;
  margin-bottom: 20px;
  text-indent: 10px;
}

.c3 {
  background: #E9ECEF;
  padding: 15px 20px;
  border-radius: 10px;
  font-weight: bold;
  color: #495057;
  cursor: pointer;
}

.c3:hover,
.c3:active {
  background: #E9ECEF;
}

.c4 {
  background: transparent;
  padding: 15px 20px;
  border-radius: 10px;
  font-weight: bold;
  color: #868E96;
  cursor: pointer;
}

.c4:hover,
.c4:active {
  background: #E9ECEF;
}

<div>
  <aside
    class="c0"
  >
    <h2
      class="c1"
    >
      title
    </h2>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        a
      </li>
      <li
        class="c4"
      >
        b
      </li>
      <li
        class="c4"
      >
        c
      </li>
      <li
        class="c4"
      >
        d
      </li>
    </ul>
  </aside>
</div>
`;
